support-frontend/assets/pages/digital-subscription-landing/components/heroGift/heroStyles.js

```diff
@@ -0,0 +1,462 @@
+
+import { headline, titlepiece, body } from '@guardian/src-foundations/typography';
+import { brand, neutral, brandAltBackground } from '@guardian/src-foundations/palette';
+import { from } from '@guardian/src-foundations/mq';
+import { space } from '@guardian/src-foundations';
+
+export const wrapper = css`
+  position: relative;
+  background: #ededed;
+  display: flex;
+  flex-direction: column;
+  padding: 0 0 ${space[3]}px;
+
+  :before {
+    position: absolute;
+    top: 0;
+    left: 0;
+    width: 100%;
+    height: 120px;
+    background: #00568D;
+    content: '';
+  }
+
+  ${from.mobileLandscape} {
+    :before {
+      height: 200px;
+    }
+  }
+
+  ${from.tablet} {
+    :before {
+      top: -1px;
+    }
+  }
+`;
+
+export const pageTitle = css`
+  ${headline.medium({ fontWeight: 'bold' })};
+  color: ${neutral[97]};
+  z-index: 10;
+  background-color: transparent;
+  padding: ${space[3]}px;
+  align-self: center;
+
+  ${from.mobileLandscape} {
+    ${headline.large({ fontWeight: 'bold' })};
+  }
+
+  ${from.phablet} {
+    width: 100%;
+    padding: ${space[5]}px ${space[3]}px;
+  }
+
+  ${from.tablet} {
+    width: calc(100% - 40px);
+    align-self: center;
+  }
+
+  ${from.desktop} {
+    ${titlepiece.medium()}
+    max-width: calc(100% - 110px);
+    max-width: 1100px;
+    padding: ${space[5]}px ${space[4]}px;
+  }
+
+  ${from.leftCol} {
+    ${titlepiece.large()}
+    width: calc(100% - 80px);
+    max-width: 80.625rem;
+  }
+`;
+
+export const featureContainer = css`
+  position: relative;
+  display: flex;
+  flex-direction: column;
+  align-self: flex-start;
+  background-color: ${brand[300]};
+  color: ${neutral[97]};
+  padding: ${space[3]}px;
+  padding-bottom: 0;
+  width: 100%;
+
+  ${from.tablet} {
+    display: inline-flex;
+    flex-direction: row;
+    width: 100%;
+    align-self: center;
+    padding: 0 ${space[4]}px ${space[4]}px;
+  }
+
+  ${from.tablet} {
+    width: calc(100% - 40px);
+  }
+
+  ${from.desktop} {
+    justify-content: space-between;
+    max-width: calc(100% - 110px);
+    max-width: 1100px;
+  }
+
+  ${from.leftCol} {
+    width: calc(100% - 80px);
+    max-width: 80.625rem;
+    padding-bottom: 0;
+  }
+`;
+
+export const textSection = css`
+  width: 100%;
+
+  ${from.tablet} {
+    padding: ${space[1]}px 0;
+    width: 55%;
+  }
+
+  ${from.leftCol} {
+    width: 40%;
+    padding-bottom: 40px;
+  }
+`;
+
+export const giftTag = css`
+  display: flex;
+  flex-direction:column;
+`;
+
+export const toFromLines = css`
+  display: inline-flex;
+  justify-content: flex-start;
+`;
+
+export const toYouTyping = css`
+  overflow: hidden; /* Ensures the content is not revealed until the animation */
+  white-space: nowrap; /* Keeps the content on a single line */
+  letter-spacing: 0.01em; /* Adjust as needed */
+  margin-left: ${space[2]}px;
+  animation:
+    typing-to 0.7s steps(3, end);
+
+  /* This is to make it work on iPhones */
+  -webkit-animation-name: typing-to;
```

Comment:
That's a great question, and it hadn't actually occurred to me to check that. What I know is that the animations weren't working in iOS and this made them work. I will see what I can find out about vendor prefixing with emotion as it may be that there's a plugin for this.

Reply:
I've had a further look at this, and it seems that emotion has a [keyframes helper(https://emotion.sh/docs/keyframes), which probably handles the vendor prefixing. The docs don't say explicitly that it does, but most of the discussion I've seen suggests that emotion should handle vendor-prefixing out of the box.

Reply:
OK great, I've taken a look at https://emotion.sh/docs/keyframes and the keyframes helper looks like it should handle the vendor prefixing. It looks like they're using it on `dotcom-rendering` too: https://github.com/guardian/dotcom-rendering/blob/55ccdbe592a5895133b67234ea18fb81f1467e0c/src/web/components/PulsingDot.tsx#L7

I'd say this shouldn't be a blocker to releasing before testing but we should look at updating the animation to use this helper as it should tidy the code up a little.